spec/rubocop/cop/layout/multiline_assignment_layout_spec.rb

```diff
@@ -23,16 +23,46 @@
       RUBY
     end
 
+    it 'registers an offense when the rhs is on the same line in []=' do
+      expect_offense(<<~RUBY)
+        hash[:foo] = if true
+        ^^^^^^^^^^^^^^^^^^^^ Right hand side of multi-line assignment is on the same line as the assignment operator `=`.
+        end
+      RUBY
+    end
+
+    it 'registers an offense when the rhs is on the same line in setters' do
+      expect_offense(<<~RUBY)
+        foo.bar = if true
+        ^^^^^^^^^^^^^^^^^ Right hand side of multi-line assignment is on the same line as the assignment operator `=`.
+        end
+      RUBY
+    end
+
     it 'auto-corrects offenses' do
       new_source = autocorrect_source(<<~RUBY)
         blarg = if true
         end
+
+        hash[:foo] = if true
+        end
+
+        foo.bar = if true
+        end
       RUBY
 
       expect(new_source).to eq(<<~RUBY)
         blarg =
          if true
         end
+
+        hash[:foo] =
+         if true
+        end
+
+        foo.bar =
+         if true
+        end
```

Comment:
I think using `expect_correction` method will keep the tests isolated and DRY.

```diff
diff --git a/spec/rubocop/cop/layout/multiline_assignment_layout_spec.rb b/spec/rubocop/cop/layout/multiline_assignment_layout_spec.rb
index a5b1e66db..99107f6b9 100644
--- a/spec/rubocop/cop/layout/multiline_assignment_layout_spec.rb
+++ b/spec/rubocop/cop/layout/multiline_assignment_layout_spec.rb
@@ -29,6 +29,12 @@ RSpec.describe RuboCop::Cop::Layout::MultilineAssignmentLayout, :config do
         ^^^^^^^^^^^^^^^^^^^^ Right hand side of multi-line assignment is on the same line as the assignment operator `=`.
         end
       RUBY
+
+      expect_correction(<<~RUBY)
+        hash[:foo] =
+         if true
+        end
+      RUBY
     end

     it 'registers an offense when the rhs is on the same line in setters' do
@@ -37,32 +43,24 @@ RSpec.describe RuboCop::Cop::Layout::MultilineAssignmentLayout, :config do
         ^^^^^^^^^^^^^^^^^ Right hand side of multi-line assignment is on the same line as the assignment operator `=`.
         end
       RUBY
+
+      expect_correction(<<~RUBY)
+        foo.bar =
+         if true
+        end
+      RUBY
     end

     it 'auto-corrects offenses' do
       new_source = autocorrect_source(<<~RUBY)
         blarg = if true
         end
-
-        hash[:foo] = if true
-        end
-
-        foo.bar = if true
-        end
       RUBY

       expect(new_source).to eq(<<~RUBY)
         blarg =
          if true
         end
-
-        hash[:foo] =
-         if true
-        end
-
-        foo.bar =
-         if true
-        end
       RUBY
     end
```

Reply:
Much better, thanks!